### PR TITLE
Add similar movies using KNN recommendation model

### LINKED ISSUE #31: Add suggested movies to movie_details page
While on the movie details_page implement a feature to show 5 related movies. Use the KNN code for the implementation. If the movie is not present in the db show a message saying that movie is not currently available.

### PR DESCRIPTION
**Pull request by Aryan10**
## Description

Integrated the KNN-based recommendation system that suggests similar movies to a particular move based on genre with the web page to show recommendations. If movie is not present in dataset, no recommendations are shown (this may be adjusted in future)

## Related Issue

Fixes #31

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Motivation and Context

This feature allows users to see movie recommendations based on the current movie they’re viewing, enhancing user engagement.

## How Has This Been Tested?

The feature was manually tested in the development environment to verify:
- The similarity recommendations display correctly for each movie
- No errors occur when movies with no recommendations are encountered
- Template renders without layout or display issues

## Screenshots
<details>
<summary>Screenshots</summary>

Screenshot 2024-10-25 151941

Screenshot 2024-10-25 152137

Screenshot 2024-10-25 152304

Screenshot 2024-10-25 152419

Screenshot 2024-10-25 152641

</details>

## Checklist:

- [x] I have registered myself at [Contrihub website](https://sac.mnnit.ac.in/contrihub).
- [x] My code follows the code style of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] Any dependent changes have been merged and published in downstream modules.
- [ ] I have added tests to cover my changes (if applicable).
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Aryan10** [comment]
I could not find a movie_details page so I have added it to movie_list for now, is movie_details page related to #13 ? If so that issue has not been closed yet.

**Comment by Aryan10** [comment]
@eleensmathew Please take a look into this.

**Comment by eleensmathew** [comment]
Hey i merged the movie _detail page so if you have the time, can you include it there? I will give the bonus

**Comment by eleensmathew** [comment]
Also pls send screen shots of the feature

File: user/views.py [hunk 1]
```diff
@@ -111,11 +113,31 @@ def add_movie(request):
...
-    return render(request, 'movie_list.html', {'movies': movies})
+
+    # Fetching similar movies from KNN model
+    predicted_similar_movies = {}
+    for movie in movies:
+        try:
+            similar_movies_df = get_similar_movies(movie.title)
+            similar_movies = []
+            for index, row in similar_movies_df.iterrows():
+                similar_movies.append(row['title'])
```
Comment (eleensmathew) [inline user/views.py:177]: can you call some API that displays the suggested movie picture ?

Comment (Aryan10) [reply user/views.py:177]: Do you mean the pictures for suggested movies?

**Comment by Aryan10** [comment]
@eleensmathew I have already included the screenshots for the current working but ok I will add the feature in movie_details page soon and get back to you.

File: user/views.py (hunk 1 above)
Comment (eleensmathew) [reply user/views.py:177]: Yup

Comment (Aryan10) [reply user/views.py:177]: Is there an existing API in the project for movie images or should I look for one?

Comment (eleensmathew) [reply user/views.py:177]: Right now there are 2 apis being used omdb and tmdb. You can probably use omdb and search with the title. When you make the change add the updated screenshots. I saw your earlier ss right now so thanks for letting me know about that

Comment (Aryan10) [reply user/views.py:177]: Ok thanks for the clarification, I will add the pictures and integrate the suggestions in the movie_details page.

**Comment by Aryan10** [comment]
Closing it as I have made a new branch, please look at  #35

**Pull request closed without merging**